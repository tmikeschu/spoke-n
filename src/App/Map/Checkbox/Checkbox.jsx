import React, { Component } from 'react';
import { categoryLabels } from '../category_data';

export default class Checkbox extends Component {
  categoryLabels() {
    return categoryLabels;
  }

  render() {
    const category = this.props.category;

    return(
      <article className="checkbox">
        <input 
          type="checkbox"
          value={category}
          onChange={event => this.props.filterPins(event)}
        />
        <label>{ this.categoryLabels()[category] }</label>
      </article>
    );
  }
}

