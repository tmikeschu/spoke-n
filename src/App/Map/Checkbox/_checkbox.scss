.checkboxes {
  display: flex;
  align-items: center;
  @include map-info-border;
  padding: 0 1%;
  justify-content: space-between;
  >article {
    width: 40%;
  }

  @media (max-width: 640px) {
    font-size: .5em;
    flex-direction: column;
    >article {
      width: 100%;
    }
  }
}

