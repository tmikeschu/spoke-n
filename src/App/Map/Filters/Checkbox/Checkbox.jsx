import React from 'react'
import PropTypes from 'prop-types'
import { categoryLabels } from '../../category_data'

const Checkbox = ({ category, filterPins }) => (
  <article className="checkbox">
    <input 
      type="checkbox"
      value={category}
      onChange={event => filterPins(event)}
    />
    <label>{ categoryLabels[category] }</label>
  </article>
)

Checkbox.propTypes = {
  category: PropTypes.string.isRequired,
  filterPins: PropTypes.func.isRequired
}

export default Checkbox

