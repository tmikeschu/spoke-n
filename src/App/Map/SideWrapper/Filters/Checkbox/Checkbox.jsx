import React from 'react'
import PropTypes from 'prop-types'
import { categoryLabels } from '../../../category_data'

const Checkbox = ({ category, filterPins, pinFilters }) => (
  <article className="checkbox">
    <input 
      type="checkbox"
      value={category}
      onChange={event => filterPins(event)}
      checked={pinFilters.includes(category)}
    />
    <label>{ categoryLabels[category] }</label>
  </article>
)

Checkbox.propTypes = {
  category: PropTypes.string.isRequired,
  filterPins: PropTypes.func.isRequired,
  pinFilters: PropTypes.array.isRequired
}

export default Checkbox

