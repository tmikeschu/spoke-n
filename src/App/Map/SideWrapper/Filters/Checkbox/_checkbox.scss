.Checkboxes {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  >article {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    width: 100%;
    article {
      margin: 8px 0;
      width: 50%;
      display: flex;
      align-items: center;
      label {
        width: 100%;
        &:hover, &:focus {
          color: $color2;
          cursor: pointer;
        }
      }
    }
  }
}

