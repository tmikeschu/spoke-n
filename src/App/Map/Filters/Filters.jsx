import React from 'react'
import PropTypes from 'prop-types'
import Checkbox from "./Checkbox/Checkbox"
import * as FilterHelper from "./FilterHelper/FilterHelper"

const Filters = props => {
  const { categories, pinFilters, setFilters } = props

  const filterPins = event => {
    const filters = FilterHelper.filterPins(event, pinFilters)
    setFilters(filters)
  }


  const checkboxes = ["", "DISPLAYNONE"].concat(categories)
    .map((category, i) => (
      <Checkbox key={i} category={category} filterPins={filterPins} />)
    )

  return(
    <article className="checkboxes" >
      <h4>Filter Suggestions</h4>
      <article>
        { checkboxes }
      </article>
    </article>
  )
}

Filters.propTypes = {
  categories: PropTypes.array.isRequired,
  pinFilters: PropTypes.array.isRequired,
  setFilters: PropTypes.func.isRequired
}

export default Filters
