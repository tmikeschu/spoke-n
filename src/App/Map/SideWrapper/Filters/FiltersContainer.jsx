import { connect } from "react-redux"
import { bindActionCreators } from "redux"
import * as actions from "../../../../Redux/actions"
import Filters from "./Filters"

const mapStateToProps = state => ({
  pinFilters: state.map.pinFilters
})

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(actions, dispatch)
})

export default connect(mapStateToProps, mapDispatchToProps)(Filters)
