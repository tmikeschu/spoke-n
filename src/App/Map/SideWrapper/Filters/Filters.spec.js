import React from 'react'
import { shallow } from "enzyme"
import Filters from "./Filters"
import Checkbox from "./Checkbox/Checkbox"

describe("<Filters />", () => {
  const filters = shallow(
    <Filters 
      categories={["stay"]}
      pinFilters={["stay"]}
      setFilters={function setFilters() {}}
    />
  )

  it("renders without crashing", () => {
    expect(filters).toBeTruthy()
  })

  describe("snapshot", () => {
    it("is valid", () => {
      expect(filters.getNodes()).toMatchSnapshot()
    })
  })
})

