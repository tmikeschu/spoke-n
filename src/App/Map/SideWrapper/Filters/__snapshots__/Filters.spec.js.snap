// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filters /> snapshot is valid 1`] = `
Array [
  <article
    className="checkboxes"
>
    <h4>
        Filter Suggestions
    </h4>
    <article>
        <Checkbox
            category=""
            filterPins={[Function]}
            pinFilters={
                Array [
                    "stay",
                  ]
            }
        />
        <Checkbox
            category="DISPLAYNONE"
            filterPins={[Function]}
            pinFilters={
                Array [
                    "stay",
                  ]
            }
        />
        <Checkbox
            category="stay"
            filterPins={[Function]}
            pinFilters={
                Array [
                    "stay",
                  ]
            }
        />
    </article>
</article>,
]
`;
